File: P12940.py
def gcd(a, b):
    
    if b > a:
        a, b = b, a
    
    while b:
        a, b = b, a%b
    
    return a

def solution(n, m):
    
    answer = []
    d = gcd(n, m)
    answer = [d, n*m//d]
    return answer

File: P12940.java
public class P12940 {
    public int[] solution(int n, int m) {
        int g = gcd(n, m);
        int[] answer = {g, n*m/g};
        return answer;
    }
    
    public int gcd(int a, int b) {
        if (b > a) {
            int tmp = a;
            a = b;
            b = tmp;
        }
        
        while (b > 0) {
            int tmp = a;
            a = b;
            b = tmp%b;
        }
        
        return a;
    }
}
